src/consensus/consensus.h

```diff
@@ -6,10 +6,24 @@
 #ifndef BITCOIN_CONSENSUS_CONSENSUS_H
 #define BITCOIN_CONSENSUS_CONSENSUS_H
 
+#include <stdint.h>
+
+static const uint64_t MEG_FORK_TIME = 1447200000; // 11 Nov 2015 00:00:00 UTC
+
 /** The maximum allowed size for a serialized block, in bytes (network rule) */
-static const unsigned int MAX_BLOCK_SIZE = 1000000;
+inline unsigned int MaxBlockSize(uint64_t nBlockTimestamp) {
+    // 1MB blocks until 11 Nov 2015, then 2MB
+    return (nBlockTimestamp < MEG_FORK_TIME ? 1000*1000 : 2*1000*1000);
```

Comment:
The old and new sizes should also be in Consesnsus::Params. MaxBlockSize should therefore take a height and a const Consesnsus::Params& as parameters. The same goes for MaxBlockSigops().